src/back-end/lib/db.ts

```diff
@@ -1400,11 +1421,21 @@ export const readOneCWUProposal = tryDb<[Id, Session], CWUProposal | null>(async
       .where({ proposal: result.id })
       .select('file')).map(row => row.file);
 
-    const rawProposalStasuses = await connection<RawCWUProposalHistoryRecord>('cwuProposalStatuses')
-      .where({ proposal: result.id })
-      .orderBy('createdAt', 'desc');
+    // Include proposal history for admins/opportunity owners
+    if (await readCWUProposalHistory(connection, session, result.opportunity)) {
+      const rawProposalStasuses = await connection<RawCWUProposalHistoryRecord>('cwuProposalStatuses')
+        .where({ proposal: result.id })
+        .orderBy('createdAt', 'desc');
 
-    result.history = await Promise.all(rawProposalStasuses.map(async raw => await rawCWUProposalHistoryRecordToCWUProposalHistoryRecord(connection, session, raw)));
+      result.history = await Promise.all(rawProposalStasuses.map(async raw => await rawCWUProposalHistoryRecordToCWUProposalHistoryRecord(connection, session, raw)));
+    }
+
+    // Fetch submittedAt date if it exists (get most recent submitted status)
+    result.submittedAt = (await connection<{ submittedAt: Date }>('cwuProposalStatuses')
+      .where({ proposal: result.id, status: CWUProposalStatus.Submitted })
+      .orderBy('createdAt', 'desc')
+      .select('createdAt as submittedAt')
+      .first())?.submittedAt;
```

Comment:
This should be abstracted to a helper function within `db.ts` since it's re-used

Reply:
Extracted to `getCWUProposalSubmittedAt` in `db.ts`